### DISCUSSION TITLE
Azure AD Oauth2 SSO integrate issue

### DISCUSSION THREAD

**Discussion by huypt-ws** [sni/thruk#1303] [Q&A]
Hi all,
I'm trying to integrate Azure AD Oauth2 login method to Thruk but encountered an issue. Below is my configuration in Thruk:
<auth_oauth>
  <provider name>
    login         = "Azure AD"
    client_id     = xxxxx
    client_secret = xxxxx
    scopes        = openid profile email
    auth_url      = https://login.microsoftonline.com/Tenant-ID/oauth2/v2.0/authorize
    token_url     = https://login.microsoftonline.com/Tenant-ID/oauth2/v2.0/token
    api_url       = https://graph.microsoft.com/v1.0/me
    login_field   = email
  </provider>
</auth_oauth>

I'm getting error from Azure portal as below:
"Sorry, but we’re having trouble signing you in.
AADSTS900971: No reply address provided."

RedirectURI is set in Azure AD already as: https://fqdn/thruk/cgi-bin/login.cgi

I really have no idea what could cause this problem nor knowing how to debug on this issue. May I have a change for any help here as I couldn't find anymore information anywhere else. Thank you very much.

**Comment by pbiering** [comment]
I got it working using

```
<auth_oauth>
  <provider name>
    login         = "Login with $COMPANY SSO"
    icon          = "uil uil-enter"
    client_id     = $APPLICATIONID
    client_secret = $SECRET
    scopes        = openid profile email
    auth_url      = https://login.microsoftonline.com/$TENANTID/oauth2/v2.0/authorize
    token_url     = https://login.microsoftonline.com/$TENANTID/oauth2/v2.0/token
    login_field   = preferred_username
  </provider>
</auth_oauth>
```

And not configuring the `api_url` at all (for now).

And the callback URL which is sent by Thruk is in whitelist on $APPLICATIONID. 

@sni : potentially the example section can be enrichted here:

https://www.thruk.org/documentation/oauth2.html

**Reply by sni** [reply]
i'll add it to the example.

**Comment by pbiering** [comment]
Did you forgot to wrap your config between:
```
  <provider NAME>
...
  </provider>
```
